app/src/main/java/com/android/feedme/model/data/IngredientsRepository.kt

```diff
@@ -101,39 +102,25 @@ class IngredientsRepository(val db: FirebaseFirestore) {
    */
   fun getIngredients(
       ingredientIds: List<String>,
-      onSuccess: (List<IngredientMetaData>) -> Unit,
+      onSuccess: (List<Ingredient>) -> Unit,
       onFailure: (Exception) -> Unit
   ) {
-    val ingredients = mutableListOf<IngredientMetaData>()
+    val ingredientIds = ingredientIds.filter { it.isNotEmpty() }
     if (ingredientIds.isEmpty()) {
-      onSuccess(ingredients) // Immediate success for empty input
+      onSuccess(emptyList()) // Immediate success for empty input
       return
     }
-
-    var completedRequests = 0
-    var hasFailed = false
-
-    ingredientIds.forEach { id ->
-      db.collection(collectionPath)
-          .document(id)
-          .get()
-          .addOnSuccessListener { documentSnapshot ->
-            val ingredient = documentSnapshot.toObject(Ingredient::class.java)
-            if (ingredient != null) {
-              ingredients.add(IngredientMetaData(1.0, MeasureUnit.CUP, ingredient))
-            }
-            completedRequests++
-            if (completedRequests == ingredientIds.size && !hasFailed) {
-              onSuccess(ingredients)
-            }
-          }
-          .addOnFailureListener { exception ->
-            if (!hasFailed) {
-              hasFailed = true
-              onFailure(exception) // Report failure on the first occurrence
-            }
-          }
-    }
+    db.collection(collectionPath)
+        .whereIn(FieldPath.documentId(), ingredientIds)
+        .get()
+        .addOnSuccessListener { querySnapshot ->
+          val ingredients =
+              querySnapshot.documents.mapNotNull { documentSnapshot ->
+                documentSnapshot.toObject(Ingredient::class.java)
+              }
+          onSuccess(ingredients)
+        }
+        .addOnFailureListener { exception -> onFailure(exception) }
```

Comment:
Good fix, so you're getting the `Ingredients` directly instead of `IngredientMetaData`.

Reply:
Yes